### ISSUE TITLE
Simply encoding SQL in base64 for transmission will not bypass the WAF gateway

### ISSUE THREAD

**Issue by miemieYaho** [openobserve/openobserve#3557]
### Description

Simply encoding SQL in base64 for transmission will not bypass the WAF gateway.

### Proposed solution

Pass a salt using environment variables and then obfuscate the SQL.

### Alternatives considered

nothing

**Comment by hengfeiyang** [comment]
Which API request was blocked, please give some detail.

**Comment by miemieYaho** [comment]
webUI
![image](https://github.com/openobserve/openobserve/assets/28334419/1543b863-7f3c-4329-bc75-ebfb98f389c0)
![image](https://github.com/openobserve/openobserve/assets/28334419/3f83af19-8dce-4fdf-9e10-4103a0b499bb)
I am certain that this firewall will decode base64 and recognize it as SQL.

**Comment by miemieYaho** [comment]
![image](https://github.com/openobserve/openobserve/assets/28334419/9f8482ec-80f2-4dc9-a202-200aeaf19737)
![image](https://github.com/openobserve/openobserve/assets/28334419/ce4f13fd-98af-49a3-9fc5-8ebd0b966da4)
I set ZO_UI_SQL_BASE64_ENABLED=true, and this WAF firewall will decode the Base64 and then intercept it.

**Comment by hengfeiyang** [comment]
@miemieYaho i just fixed it, haven't make the release, do you use docker? then i can give you a test version.

**Comment by miemieYaho** [comment]
yes,i use docker

**Comment by hengfeiyang** [comment]
Can you try this version:
```
public.ecr.aws/zinclabs/openobserve-dev:v0.10.6-rc1-f0cd361
```
with the ENV:
```
ZO_UI_SQL_BASE64_ENABLED = true
```

**Comment by miemieYaho** [comment]
This WAF firewall will inspect the entire request body. Even if it is simply base64 encoded, it will still be decoded and inspected.
![image](https://github.com/openobserve/openobserve/assets/28334419/5c6fd03c-677d-4352-a2ef-144e6627bc3c)
![image](https://github.com/openobserve/openobserve/assets/28334419/c8d5bc63-d256-4e9c-a505-05d9af64c98f)

**Comment by miemieYaho** [comment]
To prevent the WAF from decoding, the simplest method currently is to perform some obfuscation. For example, insert an 'a' every 5 characters in the base64 encoded string

**Comment by hengfeiyang** [comment]
```
To prevent the WAF from decoding, the simplest method currently is to perform some obfuscation. For example, insert an 'a' every 5 characters in the base64 encoded string
```
This is a function, but i don't think this is a good solution for us, it is a hacking solution.

will keep track it.

**Comment by miemieYaho** [comment]
Then you can consider using symmetric encryption, with the key specified in an environment variable, such as AES
or change the request body like this(I am using this method to bypass the WAF):
```json
{
    "table": "target_table",
    "where":"id = 123 and age > 2",
    "order" :"created_at desc"
}
```

**Comment by miemieYaho** [comment]
Can you implement symmetric encryption, such as AES or others?
Similar to the environment variable`ZO_UI_SQL_BASE64_ENABLED`, create one called`ZO_UI_SQL_AES_SECRET` to specify the AES key

**Comment by hengfeiyang** [comment]
@bjp232004 @prabhatsharma @oasisk What do you think about this case? add AES not so simple like base64. should we do that?

**Comment by prabhatsharma** [comment]
Makes sense. We can add the problem to our backlog. We can explore some alternatives. symmetric encryption is one alternative. I think simply prepending the base64 encoded text with 2 known characters e.g. `ab` will also solve it. @miemieYaho Can you check this and confirm.

**Comment by miemieYaho** [comment]
I've tried it, and adding `ab` before the Base64 encoded text can bypass the WAF @prabhatsharma

**Comment by hengfeiyang** [comment]
It sounds we can create an custom base64 encoding like `base46` to mixed the encoding chars. i will try it later.
